number_parser/parser.py

```diff
@@ -1,65 +1,67 @@
 import re
 import string
+import os
+import json
 # The initial global dictionaries that have been declared below will be changed,
 # should be easily derived automatically from CLDR data/other source for each of the languages.
 
 # We will need to have all values from 1-100 in a dictionary for some languages like hindi while others
 # english / french build 2 digit numbers using MTEN dictionary values.
 # (Difference observable in hi.json and en.json in numeral_language_data)
 
-MULTIPLIERS = {
-    "thousand": 1000,
-    "thousands": 1000,
-    "million": 1000000,
-    "millions": 1000000,
-    "billion": 1000000000,
-    "billions": 1000000000,
-    "trillion": 1000000000000,
-    "trillions": 1000000000000,
-}
-
-VALID_TOKENS_IN_NUMBERS = ["and", "-"]
-
-UNITS = {
-    word: value
-    for value, word in enumerate(
-        "one two three four five six seven eight nine".split(), 1
-    )
-}
-
-STENS = {
-    word: value
-    for value, word in enumerate(
-        "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split(), 10
-    )
-}
-
-MTENS = {
-    word: value * 10
-    for value, word in enumerate(
-        "twenty thirty forty fifty sixty seventy eighty ninety".split(), 2
-    )
-}
-
-HUNDRED = {"hundred": 100, "hundreds": 100}
-
-ALL_WORDS = {**UNITS, **STENS, **MTENS, **HUNDRED, **MULTIPLIERS}
-BASE_NUMBERS = {**UNITS, **STENS}
+LANGUAGE_DIRECTORY = "../number_parser/translation_data_merged/"
+UNIT_NUMBERS = {}
+BASE_NUMBERS = {}
+MTENS = {}
+MHUNDREDS = {}
+MULTIPLIERS = {}
+VALID_TOKENS_IN_NUMBERS = []
+ALL_WORDS = {}
+ALL_BASE_WORDS = {}
+
+def populate_dictionaries(lang):
+    file_name = lang + ".json"
+    fpath = os.path.join(LANGUAGE_DIRECTORY, file_name)
+
+    with open(fpath, 'r') as lang_file:
+        lang_data = json.load(lang_file)
```

Comment:
Reading from files is costly and affects the performance, so we should avoid it when it’s not necessary. Apart from that, the file paths are a pain. As you can see, now to make the tests pass, you need to be inside `number-parser/number_parser` and run them like this: ` pytest ../tests` (of course, `tox` doesn’t work).

Obviously, we need to read files to generate the “merged” result, but this result could be saved as a `.py` file to avoid needing to read it from a file and just import it.

To do this, I think that you won’t need to change too much the current code. You just need to adapt the script `get_cldr_numeral_data.py` to generate `py` files instead of `json` files.

You don't need to put the language-specific information in the constants (`UNIT_NUMBERS`, `MTENS`...). Instead of creating this `populate_dictionaries()` function, it would be better to create a `get_language_data(lang)` and just import the language data from its specific file. And then, instead of calling the `populate_dictionaries(lang)` function inside `parse_number` and `parse`, you can just do: `language_data = get_language_data(lang)` and then call the `number_builder` and the `check_validity` with an extra argument called `language_data`.

This last step can be fixed if we move to a class, but I prefer you to follow this and fix all the other issues and then we will migrate this code to a class. :slightly_smiling_face: